### ISSUE TITLE
[JENKINS-51253] jenkins_2.107.3_all.deb depends on java being installed but no longer has this explicit dependency

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/packaging#650]
<p>It seems that jenkins_2.107.3_all.deb no longer has an explicit dependency on default-jre-headless or java8-runtime-headless, which means jenkins installations are easily broken if "apt-get autoremove" is run (and it's generally good practice for debian/ubuntu admins to run this command periodically).</p>

<p>jenkins_2.107.2_all.deb and earlier had the correct dependency.</p>

<p>There is a comment on <a href="https://pkg.jenkins.io/debian/" class="external-link" target="_blank" rel="nofollow noopener">https://pkg.jenkins.io/debian/</a> which suggests that this is intentional:</p>

<blockquote>
<p>You will need to explicitly install a Java runtime environment, <b>because Jenkins does not work with Java 9</b>, this is the safest way to ensure your system ends properly configured. Adding an explicit dependency requirement on Java could force installation of undesired versions of the JVM. Check <a href="https://issues.jenkins.io/browse/JENKINS-40689" title="Java 11 Phase 4: Deprecate support for Java 8" class="issue-link" data-issue-key="JENKINS-40689"><del>JENKINS-40689</del></a> for more details about Jenkins and Java 9 compatibility.</p></blockquote>

<p> ...but this seems wrong- the package should just depend on java8 packages specifically.</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mbmop">mbmop</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-51253" target="_blank">jenkins_2.107.3_all.deb depends on java being installed but no longer has this explicit dependency</a></i></summary>
<i><ul>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>component(s)</b>: packaging
<li><b>label(s)</b>: regression
<li><b>resolution</b>: Won't Fix
<li><b>resolved</b>: 2022-01-27T05:19:08+00:00
<li><b>votes</b>: 1
<li><b>watchers</b>: 8
<li><b>imported</b>: 2025-11-26
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>It seems that jenkins_2.107.3_all.deb no longer has an explicit dependency on default-jre-headless or java8-runtime-headless, which means jenkins installations are easily broken if "apt-get autoremove" is run (and it's generally good practice for debian/ubuntu admins to run this command periodically).</p>

<p>jenkins_2.107.2_all.deb and earlier had the correct dependency.</p>

<p>There is a comment on <a href="https://pkg.jenkins.io/debian/" class="external-link" target="_blank" rel="nofollow noopener">https://pkg.jenkins.io/debian/</a> which suggests that this is intentional:</p>

<blockquote>
<p>You will need to explicitly install a Java runtime environment, <b>because Jenkins does not work with Java 9</b>, this is the safest way to ensure your system ends properly configured. Adding an explicit dependency requirement on Java could force installation of undesired versions of the JVM. Check <a href="https://issues.jenkins.io/browse/JENKINS-40689" title="Java 11 Phase 4: Deprecate support for Java 8" class="issue-link" data-issue-key="JENKINS-40689"><del>JENKINS-40689</del></a> for more details about Jenkins and Java 9 compatibility.</p></blockquote>

<p> ...but this seems wrong- the package should just depend on java8 packages specifically.</p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>debian/ubuntu</code></li></ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/oleg_nenashev.png" title="oleg_nenashev's avatar" /> oleg_nenashev:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-51253?focusedId=337521&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-337521">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>FYI <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olivergondza" class="user-hover" rel="olivergondza">olivergondza</a> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kohsuke" class="user-hover" rel="kohsuke">kohsuke</a> . Likely this is a regression due to the packaging issue</p>
</pre>
</details>

</ul>
</details>

<p>FYI <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olivergondza" class="user-hover" rel="olivergondza">olivergondza</a> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kohsuke" class="user-hover" rel="kohsuke">kohsuke</a> . Likely this is a regression due to the packaging issue</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>mbmop:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-51253?focusedId=337523&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-337523">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olivergondza" class="user-hover" rel="olivergondza">olivergondza</a>: correct.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=olivergondza" class="user-hover" rel="olivergondza">olivergondza</a>: correct.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-51253?focusedId=337608&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-337608">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Caused by <a href="https://github.com/jenkinsci/packaging/commit/f5efeb3309a1f17193e05442201eeff01fb51e5b#diff-385242a38c7d9464b1f85b37f1753721" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/packaging/commit/f5efeb3309a1f17193e05442201eeff01fb51e5b#diff-385242a38c7d9464b1f85b37f1753721</a> for <a href="https://issues.jenkins.io/browse/JENKINS-45287" title="Debian packaging should not be satisfied by Java 9" class="issue-link" data-issue-key="JENKINS-45287"><del>JENKINS-45287</del></a> which has long been used, just not for the 2.107 line until 2.107.3.</p>

<p>If there's a sane way to specify "only Java 8", ideally in a vendor independent manner (e.g. work on Ubuntu or whoever else uses debs), then this should be trivial to fix.</p>
</pre>
</details>

</ul>
</details>

<p>Caused by <a href="https://github.com/jenkinsci/packaging/commit/f5efeb3309a1f17193e05442201eeff01fb51e5b#diff-385242a38c7d9464b1f85b37f1753721" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/packaging/commit/f5efeb3309a1f17193e05442201eeff01fb51e5b#diff-385242a38c7d9464b1f85b37f1753721</a> for <a href="https://issues.jenkins.io/browse/JENKINS-45287" title="Debian packaging should not be satisfied by Java 9" class="issue-link" data-issue-key="JENKINS-45287"><del>JENKINS-45287</del></a> which has long been used, just not for the 2.107 line until 2.107.3.</p>

<p>If there's a sane way to specify "only Java 8", ideally in a vendor independent manner (e.g. work on Ubuntu or whoever else uses debs), then this should be trivial to fix.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>tonimueller:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-51253?focusedId=345733&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-345733">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>If you depend on java8-runtime-headless, than that's obviously too general. Therefore I suggested depending on openjdk-8-jre-headless for Debian/Stretch (don't know about Ubuntu), which will be OpenJDK 8 <em>only, no other version</em>. If you also want to accommodate other editions of Java, I don't know what to do, but I consider OpenJDK the go-to edition of Java on Debian. You then should not need to conflict with anything, just make sure you pick up the right version of Java if several versions are installed.</p>

<p>I just checked, and it looks as if openjdk-8-jre-headless is still also in unstable, which would allow for this version being installed on the next version of Debian unless someone pulls it (but I don't see why). In future versions of Debian after Buster, it is at some point to be expected that they will only ship higher versions of OpenJDK, so at that point, which is several years in the future, Jenkins should run on, say, OpenJDK 11. </p>
</pre>
</details>

</ul>
</details>

<p>If you depend on java8-runtime-headless, than that's obviously too general. Therefore I suggested depending on openjdk-8-jre-headless for Debian/Stretch (don't know about Ubuntu), which will be OpenJDK 8 <em>only, no other version</em>. If you also want to accommodate other editions of Java, I don't know what to do, but I consider OpenJDK the go-to edition of Java on Debian. You then should not need to conflict with anything, just make sure you pick up the right version of Java if several versions are installed.</p>

<p>I just checked, and it looks as if openjdk-8-jre-headless is still also in unstable, which would allow for this version being installed on the next version of Debian unless someone pulls it (but I don't see why). In future versions of Debian after Buster, it is at some point to be expected that they will only ship higher versions of OpenJDK, so at that point, which is several years in the future, Jenkins should run on, say, OpenJDK 11. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>mbmop:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-51253?focusedId=345739&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-345739">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Then how about we depend upon "openjdk-8-jre-headless | &lt;a similar version-specific oracle jre&gt;" ?</p>
</pre>
</details>

</ul>
</details>

<p>Then how about we depend upon "openjdk-8-jre-headless | <a similar version-specific oracle jre>" ?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>tonimueller:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-51253?focusedId=345748&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-345748">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>That should imho do the trick for Debian Stretch and Buster, as far as I can see at the moment, and for Jessie users if they have backports enabled. If you can depend on </p>

<p>openjdk-8-jre-headless | openjdk-7-jre-headless | &lt;similar oracle stuff&gt;</p>

<p>then you would cover all Debian versions back to Wheezy. But I don't know if Jenkins runs on OpenJDK 7. And, as said, I don't know what to depend on for Ubuntu &amp; friends, although it might well be the very same stuff.</p>
</pre>
</details>

</ul>
</details>

<p>That should imho do the trick for Debian Stretch and Buster, as far as I can see at the moment, and for Jessie users if they have backports enabled. If you can depend on </p>

<p>openjdk-8-jre-headless | openjdk-7-jre-headless | <similar oracle stuff></p>

<p>then you would cover all Debian versions back to Wheezy. But I don't know if Jenkins runs on OpenJDK 7. And, as said, I don't know what to depend on for Ubuntu & friends, although it might well be the very same stuff.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-51253?focusedId=345750&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-345750">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<blockquote><p>But I don't know if Jenkins runs on OpenJDK 7</p></blockquote>
<p>Hasn't since 2.54.</p>
</pre>
</details>

</ul>
</details>

<blockquote><p>But I don't know if Jenkins runs on OpenJDK 7</p></blockquote>
<p>Hasn't since 2.54.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>mbmop:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-51253?focusedId=345766&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-345766">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Updated <a href="https://github.com/jenkinsci/packaging/pull/107" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/packaging/pull/107</a></p>
</pre>
</details>

</ul>
</details>

<p>Updated <a href="https://github.com/jenkinsci/packaging/pull/107" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/packaging/pull/107</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-51253?focusedId=421106&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-421106">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Won't fix this</p>
</pre>
</details>

</ul>
</details>

<p>Won't fix this</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `is_duplicated_by` from Jira: <a href="https://github.com/jenkinsci/packaging/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-51226%22">JENKINS-51226</a>]</i>
